partner_contact_in_several_companies/models/res_partner.py

```diff
@@ -0,0 +1,158 @@
+        one2many might return wrong result - i.e with "attached contact"
+        removed even if it"s directly linked to a company).
+        Actually, is easier to override a dictionary value to indicate it
+        should be ignored...
+        """
+        if mode != "search" and "search_show_all_positions" in self.env.context:
+            result = self.with_context(search_show_all_positions={"is_set": False})
+        else:
+            result = self
+        return result
+
+    @api.model
+    def search(self, args, offset=0, limit=None, order=None, count=False):
+        """Display only standalone contact matching ``args`` or having
+        attached contact matching ``args``"""
+        ctx = self.env.context
+        if (
+            ctx.get("search_show_all_positions", {}).get("is_set")
+            and not ctx["search_show_all_positions"]["set_value"]
+        ):
+            args = expression.normalize_domain(args)
+            attached_contact_args = expression.AND(
+                (args, [("contact_type", "=", "attached")])
+            )
+            attached_contacts = super(ResPartner, self).search(attached_contact_args)
+            args = expression.OR(
+                (
+                    expression.AND(([("contact_type", "=", "standalone")], args)),
+                    [("other_contact_ids", "in", attached_contacts.ids)],
+                )
+            )
+        return super(ResPartner, self).search(
+            args, offset=offset, limit=limit, order=order, count=count
+        )
+
+    @api.model
+    def create(self, vals):
+        """When creating, use a modified self to alter the context (see
+        comment in _basecontact_check_context).  Also, we need to ensure
+        that the name on an attached contact is the same as the name on the
+        contact it is attached to."""
+        modified_self = self._basecontact_check_context("create")
+        if not vals.get("name") and vals.get("contact_id"):
+            vals["name"] = modified_self.browse(vals["contact_id"]).name
+        return super(ResPartner, modified_self).create(vals)
+
+    def read(self, fields=None, load="_classic_read"):
+        modified_self = self._basecontact_check_context("read")
+        return super(ResPartner, modified_self).read(fields=fields, load=load)
+
+    def write(self, vals):
+        modified_self = self._basecontact_check_context("write")
+        return super(ResPartner, modified_self).write(vals)
+
+    def unlink(self):
+        modified_self = self._basecontact_check_context("unlink")
+        return super(ResPartner, modified_self).unlink()
+
+    def _compute_commercial_partner(self):
+        """Returns the partner that is considered the commercial
+        entity of this partner. The commercial entity holds the master data
+        for all commercial fields (see :py:meth:`~_commercial_fields`)"""
+        result = super(ResPartner, self)._compute_commercial_partner()
```

Comment:
The computed method doesn't return a value
```
super(ResPartner, self)._compute_commercial_partner()
```

Reply:
Got it, I'll fix it